### PR TITLE
add test for clear() method

### PR DESCRIPTION
**Pull request by rghouzra**
## Description
this pr adds a comprehensive test for the `clear()` method of the ChunkContainer class
- Correct size reset after clearing
- Ability to reuse the container after clearing
- Consistent behavior across different container states

### CODE REVIEW

File: test/ChunkContainer_test.cpp [hunk 1]
```diff
@@ -240,6 +240,36 @@ TEST(ChunkContainer, resize)
...
+    {
+       chk.push_back(i);
+    }
+    EXPECT_THAT(chk.size(), Eq(10u));
+    for (int i = 0; i < 10; ++i) 
+    {
+        EXPECT_THAT(chk[size_t(i)], Eq(i)) << "at " << i;
+    }
+}
+
```
Comment (auto-differentiation-dev) [inline test/ChunkContainer_test.cpp:272]: Thank you for this contribution - a test for this method was indeed needed.

Could you make the test more concise, by doing the following?

1. Use `resize(20, 42)` or something like this to make it size 20 and fill with values `42`.
2. Then clear
3. Then use another `resize` call after and verify size and content. 

For checking content, you can then also use `EXPECT_THAT(chk, Each(Eq(42)))` to test that all elements in the container are equal the value 42. 

I would also verify that after `clear`, the `capacity` does not change (same as before clear).

Comment (rghouzra) [reply test/ChunkContainer_test.cpp:272]: ah yes i forgot about Each matcher

Comment (rghouzra) [reply test/ChunkContainer_test.cpp:266]: done

Comment (rghouzra) [reply test/ChunkContainer_test.cpp:266]: i think i cant use each since its use begin() and end()

Comment (auto-differentiation-dev) [reply test/ChunkContainer_test.cpp:266]: Yes, true. But for this test, which just tests `clear` method, you don't even need to check the values, just size and capacity (so just remove the checks for the values). 

But please check `capacity` just before `clear` - `resize()` affects `capacity` and that's expected. But `clear` should not affect `capacity`, only `size`.

**Review by auto-differentiation-dev — APPROVED** [review]

**Pull request merged**